packages/dds/migration-shim/src/shimHandle.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type IFluidHandle } from "@fluidframework/core-interfaces";
+import { type IShim } from "./types";
+
+/**
+ * ShimHandle is a special class to handle the fact that we are essentially creating a proxy for a DDS.
+ *
+ * ShimHandle is designed for MigrationShim and SharedTreeShim.
+ *
+ * Local handles such as the FluidObjectHandle and the SharedObjectHandle don't work as they do not properly bind the
+ * Shim's underlying DDS.
```

Comment:
I'm a bit fuzzy on handles, but couldn't we use a FluidObjectHandle if we additionally immediately bind the currentTree handle to it?

Reply:
We would need to either change `FluidObjectHandle` or change the internal runtime. What makes the handle live is `map.set("someKey", handle)`, with this API surface, it leaves much to be desired. Sure, we could call bind, but then we are already calling an API surface we don't want customers to call. The handle needs to point to the `Shim` object and